docs/semgrep-app/getting-started-with-semgrep-app.md

```diff
@@ -0,0 +1,284 @@
+
+
+#### Permissions for GitHub
+
+This section explains why Semgrep App requires specific permissions. Semgrep App requires the following permissions in order to log in through GitHub:
+
+<dl>
+    <dt>Verify your GitHub identity</dt>
+    <dd>Enables Semgrep App to read your GitHub profile data, such as your username.</dd>
+    <dt>Know which resources you can access</dt>
+    <dd>Enables Semgrep App to access and display private repositories you can scan.</dd>
+    <dt>Act on your behalf</dt>
+    <dd>Enables Semgrep App to start or stop scans on the repository and commit files for continuous integration.</dd>
+</dl>
+
+The GitHub integration app is called `semgrep-app`. It requires the following permissions:
+
+<dl>
+    <dt>Read and write permissions to <a href="https://docs.github.com/en/actions">GitHub Actions</a></dt>
+    <dd>Allows Semgrep App to cancel stuck jobs, rerun jobs, pull logs from jobs, and perform on-demand scanning.</dd>
+    <dt>Read permissions to <a href="https://docs.github.com/en/rest/reference/checks">GitHub Checks</a></dt>
+    <dd>Facilitates debugging of Semgrep App when configured out of GitHub Actions.</dd>
+    <dt>Read and write to GitHub Security Events</dt>
+    <dd>Enables integration with the GitHub Advanced Security to show Semgrep results.</dd>
+    <dt><code>.semgrepignore</code> single-file access</dt>
+    <dd>Allows debugging of requests and automatic syncing of <code>.semgrepignore</code> between the Semgrep App UI and what is checked into the repository.</dd>
+    <dt>Read/write GitHub secrets</dt>
+    <dd>Enables automatically adding of the Semgrep App Token to your repository secrets when onboarding projects. This simplifies bulk onboarding of repositories. <strong>Semgrep App cannot read the values of your existing or future secrets due to the security design of GitHub Secrets.</strong> This permission only permits Semgrep App to know the secret name and programmatically adds the Semgrep token to your repository secrets.</dd>
+</dl>
+
+### Signing in with GitLab
+
+1. Click the following link: [Sign into Semgrep](https://semgrep.dev/login?return_path=/manage/projects).
+2. Select **Sign in with GitLab**. You are redirected to the GitLab sign in page if you are not currently signed in.
+3. Sign in with your credentials.
+4. Click **Authorize**. See the GitLab documentation about [Authorized applications](https://docs.gitlab.com/ee/integration/oauth_provider.html#authorized-applications) to understand the scope of permissions requested by Semgrep.
+5. You are redirected back to Semgrep App.
+6. Click **Accept** to accept Semgrep's Terms of Service.
+7. Optional: Fill out the survey and click Complete or click Skip to omit this step.
+
+You are now signed in to Semgrep App.
+
+
+#### Permissions for GitLab
+
+Semgrep requires the following permissions (scopes) in order to function:
+
+
+<dl>
+    <dt>`api`</dt>
+    <dd>Legacy usage enables Semgrep App to retrieve repositories from GitLab.</dd>

```

Comment:
This is the last item to make sure it's absolutely correct. I was informed that we don't use api scope for what we used to use it for (i.e. ListProjects API) but still request permissions? For that reason I used "legacy" but not sure as to the accuracy of this claim. @raghavjain3 what do you think? cc @zyannes

Reply:
Should we just remove this permission and exclude it from the docs then?

Reply:
I think we can remove this permission for now since we're not using it for Gitlab. However, we do have plans to implement this feature for Gitlab. That said, I think we could eventually support an opt-in permissions model, where customers who grant the `api` scope have access to the features requiring that permission, whereas customers who do not will have a limited feature set (more manual steps for which we can provide walkthroughs/documentation). What do you think?

Reply:
Agree that we can remove the permission for now (and I'll remove it from the docs) . 
When we implement it in the future for GitLab, we can add it back.

I agree with the opt-in model as well, but will leave it to prod/eng to be the final decision owner. Cheers y'all. Thanks!

Reply:
@zyannes that sounds good! Would users be able to select individual permissions? I think with GitHub that isn't possible so wondering what it's like for Gitlab. 

Also, could we confirm with @dlukeomalley here as the PM for integrations?

Reply:
We know these permissions are overly permissive. As part of onboarding (CC @milanwilliams @TinMarkovic) we can consider reducing scope (it's a customer ask) and these docs could be updated at that time.

Reply:
Yup. It's a thing we'd love to do... but don't have now, and I'm not sure we'll have in a relevant timeframe (for these docs). I think it's fine to leave it in writing or remove it entirely, but don't hold up the docs waiting for major changes.

Reply:
Thank you @TinMarkovic! 

@dlukeomalley Just making sure I got this right:

(1) My action for this doc right now: Remove mention of api in these docs (done).
(2) @zyannes Will remove permission requirement(?)
(3) Tin/Milan/Zach will eventually return this permission with support for opt-in permissions. Docs will be updated to re-include this permission when it happens in the future.